### PR TITLE
fix: allow unswept funds to be used in boost

### PR DESCRIPTION
**Pull request by kylezs**
# Pull Request

Closes: PRO-1547

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have written sufficient tests 
- [x] I have written and tested required migrations.
- [x] I have updated documentation where appropriate.

## Summary

Figured this was easy enough to get in before 1.6.

### CODE REVIEW

File: state-chain/pallets/cf-ingress-egress/src/lib.rs [hunk 1]
```diff
@@ -1098,12 +1100,17 @@ pub mod pallet {
...
 			let booster_id = T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
+
 			ensure!(
 				T::SafeMode::get().add_boost_funds_enabled,
 				Error::<T, I>::AddBoostFundsDisabled
 			);
 			ensure!(amount > Zero::zero(), Error::<T, I>::AddBoostAmountMustBeNonZero);
 
+			// Free balances returns the unswept funds included, so if we want to be able to use
+			// those funds for boosting, we first need to sweep.
```
Comment (dandanlen) [inline state-chain/pallets/cf-ingress-egress/src/lib.rs:1111]: nit: this comment is a little out of context - it's referring to the rpc? I think simply stating that try_debit *doesn't* implicitly sweep any open positions, and so we need to do it first, would be enough.

**Review by dandanlen — COMMENTED** [review]
This will probably conflict with #5083 - would suggest adding it after that / branching off that PR.

**Review by msgmaxim — APPROVED** [review]

File: state-chain/pallets/cf-ingress-egress/src/lib.rs (hunk 1 above)
Comment (kylezs) [reply state-chain/pallets/cf-ingress-egress/src/lib.rs]: updated

**Review by dandanlen — APPROVED** [review]

**Comment by dandanlen** [comment]
(I rebased against main)

**Pull request merged**